eras/shelley/impl/src/Cardano/Ledger/Shelley/UTxO.hs

```diff
@@ -256,9 +231,15 @@ balance ::
   Era era =>
   UTxO era ->
   Core.Value era
-balance (UTxO utxo) = Map.foldl' addTxOuts zero utxo
-  where
-    addTxOuts !b out = getField @"value" out <+> b
+balance = sumTxOutValue . unUTxO
+
+-- | Sum all the value in TxOut
+sumTxOutValue ::
+  forall era f.
+  (Foldable f, Era era) =>
+  f (Core.TxOut era) ->
+  Core.Value era
+sumTxOutValue = foldMap (getField @"value")
```

Comment:
This function is called a lot. It used to be defined in terms of a foldl' ,do we have a foldl' for SplitMap. if not perhaps we should write one. Not sure how foldMap behaves. foldl' is strict and very fast.

Reply:
There is `foldMap'` too (which uses `foldl'` underneath), but all my prior benchmarks revealed that ghc is able to optimize `foldMap` just fine. In fact that is the default implementation for `sum` function in base :shrug: 

In any case, I'll change it to `foldMap'`, but at some point I'd like to write a benchmark for it, just to be 100% sure that we are using the optimal implementation, since it is a function that is used quite often.

Reply:
I added some benchmarks  for this function.

You where right, for `SplitMap` it actually made a big difference. `foldMap` was 3 times slower than then the strict variant `foldMap'`. Interesting part is that it was slower only because of bad implementation of lazy `foldr`. Once I fixed it then the difference became marginal as it is for list and `Map`. This is what I was referring to about that being my prior experience with `foldMap`. 

(NB. Definition of strict `foldMap' f = foldl' (\ acc a -> acc <> f a) mempty`)